app/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database.rb

```diff
@@ -1,2 +1,167 @@
+              :label => az[:name],
+              :value => az[:name]
+            }
+          end,
+        },
+        {
+          :component  => 'text-field',
+          :id         => 'compartment_id',
+          :name       => 'compartment_id',
+          :label      => _('Compartment ID'),
+          :isRequired => true,
+        },
+        {
+          :component  => 'text-field',
+          :id         => 'username',
+          :name       => 'username',
+          :label      => _('Admin Username'),
+          :isRequired => true,
+          :condition  => {
+            :when => 'database',
+            :is   => 'mysql',
+          },
+        },
+        {
+          :component  => 'text-field',
+          :id         => 'password',
+          :name       => 'password',
+          :label      => _('Admin Password'),
+          :isRequired => true,
+        },
+        {
+          :component  => 'text-field',
+          :id         => 'shape_name',
+          :name       => 'shape_name',
+          :label      => _('Database Shape Name'),
+          :isRequired => true,
+          :condition  => {
+            :when => 'database',
+            :is   => 'mysql',
+          },
+        },
+        {
+          :component    => 'select',
+          :name         => 'subnet_name',
+          :id           => 'subnet_name',
+          :label        => _('Cloud Subnet'),
+          :includeEmpty => true,
+          :isRequired   => true,
+          :condition    => {
+            :when => 'database',
+            :is   => 'mysql',
+          },
+          :options      => ems.cloud_subnets.map do |cs|
+            {
+              :label => cs[:name],
+              :value => cs[:name]
+            }
+          end
+        },
+        {
+          :component  => 'text-field',
+          :name       => 'cpu_cores',
+          :id         => 'cpu_cores',
+          :label      => _('CPU Cores'),
+          :type       => 'number',
+          :step       => 1,
+          :isRequired => true,
+          :condition  => {
+            :when => 'database',
+            :is   => 'oracle',
+          },
+        },
+        {
+          :component  => 'text-field',
+          :name       => 'storage',
+          :id         => 'storage',
+          :label      => _('Storage Size (in Terabytes)'),
+          :type       => 'number',
+          :step       => 1,
+          :isRequired => true,
+          :condition  => {
+            :when => 'database',
+            :is   => 'oracle',
+          },
+        },
+      ],
+    }
+  end
+
+  def self.raw_create_cloud_database(ext_management_system, options)
+    options[:database] == 'oracle' ? create_oracle_database(ext_management_system, options) : create_mysql_database(ext_management_system, options)
+  rescue => err
+    _log.error("cloud database=[#{options[:name]}], error: #{err}")
+    raise
+  end
+
+  def self.create_oracle_database(ext_management_system, options)
+    require 'oci'
```

Comment:
We shouldn't require the top-level `oci` class, see https://github.com/ManageIQ/manageiq-providers-oracle_cloud/pull/52 unfortunately the cough PR author cough didn't put the starting memory down but `require "oci"` ballooned to hundreds of MiBs of memory.

You should be able to `require "oci/database/database"` and get everything that you need, API client and Models

Reply:
Thanks, I've also updated the imports for `oci/mysql/mysql` and the spec tests